src/structlog/stdlib.py

```diff
@@ -407,15 +408,12 @@ def format(self, record):
         """
         Extract ``structlog``'s `event_dict` from ``record.msg`` and format it.
         """
+        should_format = False
         if isinstance(record.msg, dict):
-            # Both attached by wrap_for_formatter
-            logger = record._logger
-            meth_name = record._name
-
-            # We need to copy because it's possible that the same record gets
-            # processed by multiple logging formatters.  LogRecord.getMessage
-            # would transform our dict into a str.
-            ed = record.msg.copy()
+            logger, meth_name, ed = self.get_logger_name_and_msg(record)
+        elif self.is_json(record.msg):
```

Comment:
Yeah sorry, this is unfortunately no bueno. :(  We had similar things proposed before but having a parser part of logging is not acceptable.